src/lib/chawathe98_matcher.rs

```diff
@@ -666,45 +1316,62 @@ impl<T: Clone + Debug + Eq + 'static> MatchingTreesScriptor<T> for Config2 {
                       &store.from_arena.borrow(),
                       to_node,
                       &store.to_arena.borrow())
-                {
-                    if from_node_id == to_node_id {
-                        // Perfect matching
-                        store.push(*from_node, *to_node, 1, EdgeType::OK);
-                    } else {
-                        // The nodes have the same value
-                        store.push(*from_node, *to_node, 1, EdgeType::MOVE)
+                    {
+                        if from_node_id == to_node_id {
```

Comment:
This would be simpler as something like:

```rust
                            let edge_type = if from_node_id == to_node_id { EdgeType::OK } else { EdgeType::Move};
                            store.push_edge(base_pre[from_node_id],
                                            diff_pre[to_node_id],
                                            1,
                                            edge_type);
```